Sources/Tracking/CustomerIO.swift

```diff
@@ -238,6 +246,59 @@ public class CustomerIO: CustomerIOInstance {
             )
     }
 
+    // Swizzled method for APN device token.
+    @objc
+    func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
+        let token = String(apnDeviceToken: deviceToken)
+        Self.shared.registerDeviceToken(token)
+    }
+
+    // Implement method swizzling for APN device token. Swizzling will ensure that the
+    // device token gets registered without needing the developer to add call respective
+    // Customer.io SDK methods to register the device.
+    @available(iOSApplicationExtension, unavailable)
+    func swizzleDidReceiveRemoteNotification() {
+        DispatchQueue.main.async {
```

Comment:
Is there a specific reason we use `main` over here?

Reply:
My bad! I implemented this code in react native package first to check feasibility. I should bring the code out of main thread.